### PR TITLE
Sorting Algos' Description Update

### PR DESCRIPTION
**Pull request by abhishekabk5**

### CODE REVIEW

File: Sorting/QuickSort.cpp [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+    An efficient, comparison-based, in-place, divide and conquer sorting algorithm. It picks
+    an element as pivot, partitions the given array around the picked pivot, recursively
+    calls itself on those partitions and then merges them after sorting.
+    Statistically, for large data sets, it amounts to be the fastest sorting algorithm.
 
     Time complexity
     ---------------
-    O(N * log(N)), where N is the number of elements.
+    Worst case: O(N^2), where N is the number of elements.
+    Average case: O(N * log(N)).
```
Comment (faheel) [inline Sorting/QuickSort.cpp]:
```
Expected worst case: O(N * log(N)),
Absolute worst case: O(N^2),
where N is the number of elements.
```

File: Sorting/QuickSort.cpp [hunk 2]
```diff
@@ -1,11 +1,15 @@
 /*
     Quick sort
     ----------
-    An efficient, comparison-based, in-place, divide and conquer sorting algorithm.
+    An efficient, comparison-based, in-place, divide and conquer sorting algorithm. It picks
+    an element as pivot, partitions the given array around the picked pivot, recursively
+    calls itself on those partitions and then merges them after sorting.
```
Comment (faheel) [inline Sorting/QuickSort.cpp]: Quick sort doesn't "merge" the partitions.

The description should therefore be like: `It randomly picks an element as pivot, partitions the given array around the selected pivot while placing the pivot in its sorted position, and recursively calls itself on those partitions.`

File: Sorting/MergeSort.cpp [hunk 3]
```diff
@@ -1,15 +1,17 @@
-/*
+ /*
```
Comment (faheel) [inline Sorting/MergeSort.cpp]: Extra space spotted!

File: Sorting/MergeSort.cpp [hunk 4]
```diff
@@ -1,15 +1,17 @@
...
+    then merges the two sorted halves.
 
     Time complexity
     ---------------
     O(N * log(N)), where N is the number of elements.
 
     Space complexity
     ----------------
-    O(N).
+    O(N), temporary space for array used in merging.
```
Comment (faheel) [inline Sorting/MergeSort.cpp]:
```
O(N), where N is the number of elements.
This space is used for merging two halves of the array.
```

**Review by faheel — CHANGES_REQUESTED** [review]
I don't think there's a need to mention `temporary space for swapping` where the space complexity is _O(1)_.

**Comment by abhishekabk5** [comment]
I hope the changes are up to mark now.
I apologise for any inconvenience caused, I am new to open source. :)

File: Sorting/QuickSort.cpp [hunk 5]
```diff
@@ -2,14 +2,15 @@
...
+    calls itself on those partitions.
     Statistically, for large data sets, it amounts to be the fastest sorting algorithm.
 
     Time complexity
     ---------------
-    Worst case: O(N^2), where N is the number of elements.
-    Average case: O(N * log(N)).
+    Absolute Worst case: O(N^2),
+    Expected Worst case: O(N * log(N)),
+    Where N is the number of elements.
```
Comment (faheel) [inline Sorting/QuickSort.cpp]: All `W`s in lower-case.

**Review by faheel — CHANGES_REQUESTED** [review]

**Comment by faheel** [comment]
@abhishekabk5 No inconvenience caused :smile:. Keep learning!

File: Sorting/QuickSort.cpp [hunk 6]
```diff
@@ -1,11 +1,16 @@
 /*
     Quick sort
     ----------
-    An efficient, comparison-based, in-place, divide and conquer sorting algorithm.
+    An efficient, comparison-based, in-place, divide and conquer sorting algorithm. It picks
+    an element as pivot, partitions the given array around the picked pivot and then recursively
+    calls itself on those partitions.
```
Comment (faheel) [inline Sorting/QuickSort.cpp]:
```
An efficient, comparison-based, in-place, divide and conquer sorting algorithm.
It randomly picks an element as pivot, partitions the given array around the selected pivot
while placing it in its proper position, and recursively calls itself on those partitions.
```

File: Sorting/InsertionSort.cpp [hunk 7]
```diff
@@ -2,7 +2,7 @@
     Insertion sort
     --------------
     A simple sorting algorithm that builds the final sorted list of values
-    one item at a time.
+    one item at a time. Less suitable for larger lists.
```
Comment (faheel) [inline Sorting/InsertionSort.cpp]:
```
A simple in-place sorting algorithm that builds the sorted list of values one item at a time.
It is more efficient than most other quadratic-time algorithms.
```

File: Sorting/MergeSort.cpp [hunk 8]
```diff
@@ -1,15 +1,18 @@
 /*
     Merge sort
     ----------
-    An efficient, comparison-based, divide and conquer sorting algorithm.
+    An efficient, comparison-based, divide and conquer sorting algorithm. It divides
+    input array into two halves, recursively calls itself for the two halves and
+    then merges the two sorted halves.
```
Comment (faheel) [inline Sorting/MergeSort.cpp]:
```
An efficient, comparison-based, divide and conquer sorting algorithm.
It divides the input array into two halves, recursively calls itself for those halves, and then merges
the two sorted halves back.
```

File: Sorting/SelectionSort.cpp [hunk 9]
```diff
@@ -1,7 +1,9 @@
 /*
     Selection sort
     --------------
-    A simple in-place comparision sorting algorithm.
+    A simple in-place comparision sorting algorithm. It sorts an array by repeatedly
+    finding the minimum element (considering ascending order) from the unsorted
+    part and putting it at its right place.
```
Comment (faheel) [inline Sorting/SelectionSort.cpp]:
```
A simple in-place comparison-based sorting algorithm.
It sorts a list by finding the smallest / largest element (depending on the sorting order)
in the unsorted sublist, swapping it with the leftmost unsorted element, and moving the sublist
boundaries one element to the right.
```

**Review by faheel — CHANGES_REQUESTED** [review]
Here are my final thoughts on the descriptions.

**NOTE**: Put a line break after the first sentence and make sure each line is at most 100 characters.

File: Sorting/InsertionSort.cpp [hunk 10]
```diff
@@ -1,8 +1,8 @@
-/*
+        /*
```
Comment (faheel) [inline Sorting/InsertionSort.cpp]: One last thing (extra tab) :sweat_smile:

**Review by faheel — CHANGES_REQUESTED** [review]

**Review by faheel — APPROVED** [review]

**Comment by faheel** [comment]
@abhishekabk5 Thanks for your time and contribution, and congratulations on your first PR merge! :smiley::+1:

**Pull request merged**